File: LicenseManagementApi/Models/Entities/Product.cs
namespace LicenseManagementApi.Models.Entities;

public class Product
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string ProductCode { get; set; } = string.Empty;
    public string? Description { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    // Navigation properties
    public ICollection<Sku> Skus { get; set; } = new List<Sku>();
    public ICollection<License> Licenses { get; set; } = new List<License>();
}

File: vue-ui-frontend/src/types/product.ts
export interface Product {
  id: string
  name: string
  productCode: string
  version: string
  description: string
  createdAt: string
  updatedAt: string
}

export interface CreateProductDto {
  name: string
  productCode: string
  version: string
  description: string
}

export interface UpdateProductDto {
  name: string
  productCode: string
  version: string
  description: string
}
